TesasSemble/fastg_graph.py

```diff
@@ -0,0 +1,160 @@
+from TesasSemble.graph import *
+from TesasSemble.graph_components import *
+
+def fastgs_to_red_blue(path_g1,path_g2):
+    
+    """
+
+    This function takes two fastg assembly outputs. 
+    Each input should be a seperate sample derived from 
+    a seperate contiional variable, such as time. 
+
+    Parameters
+    ----------
+
+    path_g1: str - a full path to fastg1
+    path_g2: str - a full path to fastg2
+
+    Returns
+    -------
+    G: of TesasSemble.graph type
+
+    Raises
+    ------
+    ValueError
+
+    None (TODO)
+
+    References
+    ----------
+    .. [1] http://fastg.sourceforge.net/
+    
+    Examples
+    --------
+
+    >>> from TesasSemble.fastg_graph import fastgs_to_red_blue
+
+    example fastgs in tests/data/fastg_test
+    - a fist sample S1.fastg
+    - a second sample S2.fastg
+    
+    they have a overlapping edge and each has a unque edge.
+
+    >>> g1 = 'tests/data/fastg_test/S1.fastg'
+    >>> g2 = 'tests/data/fastg_test/S1.fastg'
+    >>> G = fastgs_to_red_blue(g1,g2)
+
+    """
+    adj_list_g1 = fastg_to_dict(path_g1)
+    adj_list_g2 = fastg_to_dict(path_g2)
+    
+
+    ## shared red-and-blue edges 
+    clean_string = lambda word: word.replace('>','').replace("'",'').replace(";",'')
+    map_g1_g2 = [[clean_ids(x),clean_ids(y)] 
+                 for x,v1 in read_fastg(path_g1).items() 
+                 for y,v2 in read_fastg(path_g2).items() 
+                 if v1==v2]
+    #print(map_g1_g2)
+    
+    map_g1_g2_names = {x_:y_ for x,y in map_g1_g2 
+                       for x_,y_ in zip(x.split(' '),
+                                        y.split(' '))}
+    ## make g1 only (red)
+    g1_only = [x for x in adj_list_g1 
+               if not any(x==y[0] for y in map_g1_g2)]
+    ## make g2 only (blue)
+    g2_only = [x for x in adj_list_g2 
+               if not any(x==y[1] for y in map_g1_g2)]
+    # map g2 nodes to g1 nodes
+    g2_only = [' '.join([map_g1_g2_names[y] 
+                         if y in map_g1_g2_names.keys() else y 
+                         for y in x.split(' ')]) for x in g2_only]
+
+    ## add both edges and nodes ##
+    G = RedBlueDiGraph()
+
+    # red and blue
+    nodes_ = list(set([j for i,_ in map_g1_g2 
```

Comment:
I do not understand this part. Are you making g1 as a red graph and g2 a blue one?
Red edges in this case should be those edges that are in 100% of samples (or in a % to be defined). The blue ones are those that does not meet this.